engine/apps/mattermost/models/message.py

```diff
@@ -0,0 +1,42 @@
+from django.db import models
+
+from apps.alerts.models import AlertGroup
+from apps.mattermost.client import MattermostPost
+
+
+class MattermostMessage(models.Model):
+    (
+        ALERT_GROUP_MESSAGE,
+        LOG_MESSAGE,
+    ) = range(2)
+
+    MATTERMOST_MESSAGE_CHOICES = ((ALERT_GROUP_MESSAGE, "Alert group message"), (LOG_MESSAGE, "Log message"))
+
+    post_id = models.CharField(max_length=100)
+
+    channel_id = models.CharField(max_length=100)
+
+    message_type = models.IntegerField(choices=MATTERMOST_MESSAGE_CHOICES)
+
+    alert_group = models.ForeignKey(
+        "alerts.AlertGroup",
+        on_delete=models.CASCADE,
+        related_name="mattermost_messages",
+    )
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["alert_group", "channel_id", "message_type"],
+                condition=models.Q(message_type__in=[0, 1]),
```

Comment:
Last question, why do we need the condition here? (column is not nullable, and these are the 2 only possible values)

Reply:
I think we might add other types for messages that get posted in the thread like from notifications, log message etc ?

Reply:
Ack, I think we can have the uniqueness without a condition for now, but we can revisit later.